test/stdlib/KeyPath.swift

```diff
@@ -90,6 +90,8 @@ struct ComputedB {
   var reabstracted: () -> () = {}
 }
 
+typealias Tuple<T: Equatable, U> = (S<T>, C<U>)
```

Comment:
It'd be nice to add some execution tests here to verify that applying keypaths involving tuples gives the expected results, using both concrete tuples and tuples that are dependent on generic environment.